### DISCUSSION TITLE
x-show with settimeout

### DISCUSSION THREAD

**Discussion by rickavmaniac** [alpinejs/alpine#1185] [3. Feature ideas]
Hi,

A common pattern we face everyday is showing a message and hide it after 3 sec.

It would be could if Alpine could have a shortcut for that like

**x-show.timeout.3000ms(show)**

That would be equivalent to setTimeout(() => show = false, 3000)

**Comment by ryangjchandler** [comment]
@rickavmaniac Interesting idea! For now, you could achieve this using `$watch`.

```js
$watch('nameOfProp', function (value) {
    if (value) {
        setTimeout(() => show = false, 3000)
    }
})
```

**Reply by Sambuxc** [reply]
Where would one put this? inside te x-init attribute of the element or in a script tag?

**Reply by SimoTod** [reply]
In v3, you can use also use x-effect to keep it next to the x-show directive: https://codepen.io/SimoTod/pen/PoyBgJY

**Reply by Sambuxc** [reply]
Now that is fantastic! Thanks so much ♥️

**Reply by Gangadhar500error** [reply]
<div 
                                    x-data="{ show: true }"
                                    x-init="setTimeout(() => show = false, 5000)"
                                    x-show="show"
                                    x-transition:enter="transition ease-out duration-300"
                                    x-transition:enter-start="opacity-0 transform scale-90"
                                    x-transition:enter-end="opacity-100 transform scale-100"
                                    x-transition:leave="transition ease-in duration-300"
                                    x-transition:leave-start="opacity-100 transform scale-100"
                                    x-transition:leave-end="opacity-0 transform scale-90"
                                    class="flex items-center p-4 bg-white dark:bg-zinc-800 border-2 rounded-lg my-3 mx-3"
                                >
                                    <div class="flex-shrink-0 bg-blue-500 p-2 rounded-full">
                                        
                                        <svg width="20" height="20" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
                                            <circle cx="50" cy="50" r="45" stroke="transparent" stroke-width="5" fill="white"></circle>
                                            <line x1="50" y1="25" x2="50" y2="60" stroke="blue" stroke-width="8"></line>
                                            <circle cx="50" cy="75" r="5" fill="blue"></circle>
                                        </svg>
                                    </div>
                                    <div class="ml-4 text-zinc-700 dark:text-zinc-300 font-medium">success alert</div>
                                </div>
                                this code will be work I runned same

**Reply by Gangadhar500error** [reply]
<div 
                                    x-data="{ show: true }"
                                    x-init="setTimeout(() => show = false, 5000)"
                                    x-show="show"
                                    x-transition:enter="transition ease-out duration-300"
                                    x-transition:enter-start="opacity-0 transform scale-90"
                                    x-transition:enter-end="opacity-100 transform scale-100"
                                    x-transition:leave="transition ease-in duration-300"
                                    x-transition:leave-start="opacity-100 transform scale-100"
                                    x-transition:leave-end="opacity-0 transform scale-90"
                                    class="flex items-center p-4 bg-white dark:bg-zinc-800 border-2 rounded-lg my-3 mx-3"
                                >
                                    <div class="flex-shrink-0 bg-blue-500 p-2 rounded-full">
                                        
                                        <svg width="20" height="20" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
                                            <circle cx="50" cy="50" r="45" stroke="transparent" stroke-width="5" fill="white"></circle>
                                            <line x1="50" y1="25" x2="50" y2="60" stroke="blue" stroke-width="8"></line>
                                            <circle cx="50" cy="75" r="5" fill="blue"></circle>
                                        </svg>
                                    </div>
                                    <div class="ml-4 text-zinc-700 dark:text-zinc-300 font-medium">{{ session("timelinesuccess") }}</div>
                                </div>

**Comment by ekwoka** [comment]
Generally, in the part that creates the message, it's easy enough to just set a timeout right there.

or if the element is in a template x-if or x-for, you can just do `x-init="setTimeout(() => show = false"`

**Reply by Sambuxc** [reply]
Sorry, but this isn't correct from what I needed. x-init will execute the code when the page is loading, rather than when the element is shown if using x-show.

**Reply by ekwoka** [reply]
If it's shown with x-if it will.

which is why the first part of the sentence was there.

I was providing alternatives since you didn't provide much detail at all regarding what the specifics of the implementation are, and a rearchitecting may allow a cleaner result.

**Reply by Sambuxc** [reply]
Okay thanks 🙏🏽

**Comment by Gangadhar500error** [comment]
<div 
                                    x-data="{ show: true }"
                                    x-init="setTimeout(() => show = false, 5000)"
                                    x-show="show"
                                    x-transition:enter="transition ease-out duration-300"
                                    x-transition:enter-start="opacity-0 transform scale-90"
                                    x-transition:enter-end="opacity-100 transform scale-100"
                                    x-transition:leave="transition ease-in duration-300"
                                    x-transition:leave-start="opacity-100 transform scale-100"
                                    x-transition:leave-end="opacity-0 transform scale-90"
                                    class="flex items-center p-4 bg-white dark:bg-zinc-800 border-2 rounded-lg my-3 mx-3"
                                >
                                    <div class="flex-shrink-0 bg-blue-500 p-2 rounded-full">
                                        
                                        <svg width="20" height="20" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
                                            <circle cx="50" cy="50" r="45" stroke="transparent" stroke-width="5" fill="white"></circle>
                                            <line x1="50" y1="25" x2="50" y2="60" stroke="blue" stroke-width="8"></line>
                                            <circle cx="50" cy="75" r="5" fill="blue"></circle>
                                        </svg>
                                    </div>
                                    <div class="ml-4 text-zinc-700 dark:text-zinc-300 font-medium">{{ session("timelinesuccess") }}</div>
                                </div>
